text/####-error-mitigation.md

```diff
@@ -0,0 +1,235 @@
+The `experiments` in the resulting `qobj` would have an entry in the header that specifies the stretch factor.
+For example, the `qobj` for two circuits (`circA` and `circB`) to be run with error mitigation with three stretch factors would look like
+```
+print(qobj.experiments[0])
+{
+    'config': {...},
+    'header': {'name': 'circA_ef01a', 'stretch_factor': '1.0', ...},
+    'instructions': {...}
+}
+
+print(qobj.experiments[1])
+{
+    'config': {...},
+    'header': {'name': 'circA_ef01b', 'stretch_factor': '1.1', ...},
+    'instructions': {...}
+}
+print(qobj.experiments[3])
+{
+    'config': {...},
+    'header': {'name': 'circB_ec01a', 'stretch_factor': '1.0', ...},
+    'instructions': {...}
+}
+
+print(qobj.experiments[4])
+{
+    'config': {...},
+    'header': {'name': 'circB_ec01b', 'stretch_factor': '1.1', ...},
+    'instructions': {...}
+}
+```
+Here, the instructions for experiments 0, 1, and 2 are identical since they correspond to `circA`.
+The instructions for experiments 3, 4, and 5, are also identical since they correspond to `circB`.
+Therefore, the function `assemble_circuits`, called by assemble, will be modified. 
+Currently, `assemble_circuits` contains a loop over the provided circuits:
+```
+for circuit in circuits:
+    ...
+    
+    experiments.append(QasmQobjExperiment(instructions=instructions, header=header,
+                                          config=config))
+```
+We could modify this loop to be sensitive to error mitigation
+```
+for circuit in circuits:
+    ...
+    
+    if error_mitigation == 'richardson':
+        config = QasmQobjExperimentConfig(...)
+        
+        instructions = []
+        # code to create the instructions is unchanged
+    
+        for c in stretch_factors:
+            header = QobjExperimentHeader(..., stretch_factor=c)
+        
+            experiments.append(QasmQobjExperiment(instructions=instructions, header=header,
+                                                  config=config))
+    else:
+        header = QobjExperimentHeader(..., stretch_factor=None)
+        experiments.append(QasmQobjExperiment(instructions=instructions, header=header,
+                                              config=config))
+```
+This implies that the header of the `QasmQobjExperiment` will need to be modified to include an attribute, namely the `stretch_factor`.
+
+### Backend execution
+The backend will have to be able to detect that stretch factors are present in the header of the `qobj.experiments`.
+This will then signal to the backend what type of gates to use to execute each circuit.
+
+### Result returned by the backend
+The result returned by the backend would contain the results for the circuits executed with different stretch factors.
+This result object could then be given to a Richardson error mitigation module in Qiskit Ignis that would perform the extrapolation to the zero-noise limit.
```

Comment:
```suggestion
This result object could then be given to a Richardson error mitigation module in Qiskit Ignis that would perform the extrapolation to the zero-noise limit.
This module would need to know which result is associated to which quantum circuit and stretch factor.
This may be done by adding the required stretch factor as a parameter in the header of the result. For the example above, with `circA` and `circB`, the backend may return the `mitigated_result` in which the header of each result may look like

print(mitigated_result.results[0].header)
Obj(name='circA', stretch_factor='1.0')
print(mitigated_result.results[1].header)
Obj(name='circA', stretch_factor='1.1')
print(mitigated_result.results[3].header)
Obj(name='circB', stretch_factor='1.0')
print(mitigated_result.results[4].header)
Obj(name='circB', stretch_factor='1.1')

```

Reply:
sounds good